velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp

```diff
@@ -57,6 +57,79 @@ class DateTimeFunctionsTest : public SparkFunctionBaseTest {
   }
 };
 
+TEST_F(DateTimeFunctionsTest, toUTCTimestamp) {
+  const auto toUTCTimestamp = [&](const std::string_view ts,
+                                  const std::optional<std::string> tz) {
+    auto timestamp = std::make_optional<Timestamp>(
+        util::fromTimestampString(ts.data(), ts.length()));
+    return evaluateOnce<Timestamp>("to_utc_timestamp(c0, c1)", timestamp, tz);
+  };
+  EXPECT_EQ(
+      util::fromTimestampString("2015-07-24 07:00:00"),
+      toUTCTimestamp("2015-07-24 00:00:00", "America/Los_Angeles"));
+  EXPECT_EQ(
+      util::fromTimestampString("2015-01-24 08:00:00"),
+      toUTCTimestamp("2015-01-24 00:00:00", "America/Los_Angeles"));
+  EXPECT_EQ(
+      util::fromTimestampString("2015-01-24 00:00:00"),
+      toUTCTimestamp("2015-01-24 00:00:00", "UTC"));
+  EXPECT_EQ(
+      util::fromTimestampString("2015-01-24 00:00:00"),
+      toUTCTimestamp("2015-01-24 05:30:00", "Asia/Kolkata"));
+  VELOX_ASSERT_THROW(
+      toUTCTimestamp("2015-01-24 00:00:00", "Asia/Ooty"),
+      "Asia/Ooty not found in timezone database");
+}
+
+TEST_F(DateTimeFunctionsTest, fromUTCTimestamp) {
+  const auto fromUTCTimestamp = [&](const std::string_view ts,
+                                    const std::optional<std::string> tz) {
+    auto timestamp = std::make_optional<Timestamp>(
+        util::fromTimestampString(ts.data(), ts.length()));
+    return evaluateOnce<Timestamp>("from_utc_timestamp(c0, c1)", timestamp, tz);
+  };
+  EXPECT_EQ(
+      util::fromTimestampString("2015-07-24 00:00:00"),
+      fromUTCTimestamp("2015-07-24 07:00:00", "America/Los_Angeles"));
+  EXPECT_EQ(
+      util::fromTimestampString("2015-01-24 00:00:00"),
+      fromUTCTimestamp("2015-01-24 08:00:00", "America/Los_Angeles"));
+  EXPECT_EQ(
+      util::fromTimestampString("2015-01-24 00:00:00"),
+      fromUTCTimestamp("2015-01-24 00:00:00", "UTC"));
+  EXPECT_EQ(
+      util::fromTimestampString("2015-01-24 05:30:00"),
+      fromUTCTimestamp("2015-01-24 00:00:00", "Asia/Kolkata"));
+  VELOX_ASSERT_THROW(
+      fromUTCTimestamp("2015-01-24 00:00:00", "Asia/Ooty"),
+      "Asia/Ooty not found in timezone database");
+}
+
+TEST_F(DateTimeFunctionsTest, toFromUTCTimestamp) {
```

Comment:
This test seems redundant. Any particular reason to include it?

Reply:
This test takes in a timestamp and does a to and from UTC conversion essentially testing to see that the 2 functions are the inverse of each other.